### PR TITLE
added javadox documentation for Arduino_MachineControl.h

### PR DESCRIPTION
**Pull request by ErnestoELC**

### CODE REVIEW

File: src/Arduino_MachineControl.h [hunk 1]
```diff
@@ -64,9 +98,21 @@ class COMMClass {
 		digitalWrite(PH_13, LOW);
 	}
 
+	 /**
+	 * Set the CAN transciever in Normal mode. In this mode, the transceiver
```
Comment (per1234) [inline src/Arduino_MachineControl.h:102]:
```suggestion
	 * Set the CAN transceiver in Normal mode. In this mode, the transceiver 
```

File: src/Arduino_MachineControl.h [hunk 2]
```diff
@@ -64,9 +98,21 @@ class COMMClass {
...
+	 /**
+	 * Set the CAN transciever in Normal mode. In this mode, the transceiver 
+	 * can transmit and receive data via the bus lines CANH and CANL.
+	 */
 	void enableCAN() {
 		can_disable = 0;
 	}
+
+	 /**
+	 * Set the CAN transciever in standby (low power) mode. In this mode the
```
Comment (per1234) [inline src/Arduino_MachineControl.h:110]:
```suggestion
	 * Set the CAN transceiver in standby (low power) mode. In this mode the
```

File: src/Arduino_MachineControl.h [hunk 3]
```diff
@@ -117,6 +173,10 @@ class AnalogInClass {
         return value;
     }
 
+	 /**
+	 * Configure the input resistor deviders to have a ratio of 0.28.
```
Comment (per1234) [inline src/Arduino_MachineControl.h:177]:
```suggestion
	 * Configure the input resistor dividers to have a ratio of 0.28.
```

File: src/Arduino_MachineControl.h [hunk 4]
```diff
@@ -134,6 +194,10 @@ class AnalogInClass {
 		ch2_in4 = 1;
 	}
 
+	 /**
+	 * Enable a 120 ohm resitor to GND to convert the 4-20mA sensor currents to voltage.
```
Comment (per1234) [inline src/Arduino_MachineControl.h:198]:
```suggestion
	 * Enable a 120 ohm resistor to GND to convert the 4-20mA sensor currents to voltage. 
```

File: src/Arduino_MachineControl.h [hunk 5]
```diff
@@ -151,6 +215,10 @@ class AnalogInClass {
 		ch2_in4 = 0;
 	}
 
+	 /**
+	 * Enable a 100K resitor in series with the reference voltage.
```
Comment (per1234) [inline src/Arduino_MachineControl.h:219]:
```suggestion
	 * Enable a 100K resistor in series with the reference voltage.	 
```

File: src/Arduino_MachineControl.h [hunk 6]
```diff
@@ -420,14 +512,35 @@ extern EncoderClass encoders;
...
+/**
+ * The ProgrammableDIOClass is used to initialize the IOExpanders and configure the 
+ * thermal shutdown mode of the high side switches.
+ */
 class ProgrammableDIOClass : public ArduinoIOExpanderClass {
 public:
+
+	/**
+	 * Test connection with the IOExpander and set all the pins to the default mode. 
+	 * @return True if OK, False if fault
```
Comment (per1234) [inline src/Arduino_MachineControl.h:525]:
```suggestion
	 * @return true if OK, false if fault
```

File: src/Arduino_MachineControl.h [hunk 7]
```diff
@@ -470,14 +611,22 @@ extern DigitalOutputsClass digital_outputs;
 
 class ProgrammableDINClass : public ArduinoIOExpanderClass {
 public:
+	/**
+	 * Test connection with the IOExpander and set all the pins to the default mode. 
+	 * @return True if OK, False if fault
```
Comment (per1234) [inline src/Arduino_MachineControl.h:616]:
```suggestion
	 * @return true if OK, false if fault
```

File: src/Arduino_MachineControl.h [hunk 8]
```diff
@@ -488,21 +637,41 @@ class RtcControllerClass : public PCF8563TClass {
...
+	 * Disable power to USBA VBUS.  
+	 */
 	void powerDisable() {
 		power = 1;
 	}
 
+	/**
+	 * Flag to indicate overcurrent, overtemperature, or reverse−voltage conditions on the USBA VBUS. 	 
+	 * Active−low open−drain output.
+	 * @return True if OK, False if fault
```
Comment (per1234) [inline src/Arduino_MachineControl.h:673]:
```suggestion
	 * @return true if OK, false if fault
```

**Review by per1234 — CHANGES_REQUESTED** [review]

File: src/utility/RTC/PCF8563T.cpp [hunk 9]
```diff
@@ -28,7 +28,7 @@
 #define PCF8563T_MONTHS_REG     0x07
 #define PCF8563T_YEARS_REG      0x08
 
-// allarm management
+// allarm managemet
```
Comment (per1234) [inline src/utility/RTC/PCF8563T.cpp:31]:
```suggestion
// alarm management
```

File: src/utility/RTC/PCF8563T.cpp [hunk 10]
```diff
@@ -49,10 +49,20 @@
...
+ *  Object constructor
+ *  
+ */   
 PCF8563TClass::PCF8563TClass()
 {
 }
 
+/**
+ *  Start the communication with the RTC
+ *  Initialize I2C (Wire1) bus and check if the chip is connected trhoug sending an ACK on the I2C bus.
```
Comment (per1234) [inline src/utility/RTC/PCF8563T.cpp:62]:
```suggestion
 *  Initialize I2C (Wire1) bus and check if the chip is connected by sending an ACK on the I2C bus.
```

**Review by per1234 — DISMISSED** [review]

**Comment by sebromero** [comment]
@ErnestoELC Can you have a look at Per's suggestions?

**Review by marqdevx — APPROVED** [review]

**Review by marqdevx — APPROVED** [review]

**Review by marqdevx — APPROVED** [review]

**Review by marqdevx — APPROVED** [review]

**Comment by marqdevx** [comment]
@per1234 Thanks for the suggestions!
Now I cannot remove the changes requested, how should I?

Thanks again!! :D

**Review by marqdevx — APPROVED** [review]

**Comment by per1234** [comment]
@marqdevx I have now dismissed the requests as resolved. Thanks!

**Pull request merged**